packages/api/endpoints/elasticsearch.js

```diff
@@ -0,0 +1,186 @@
+    }
+
+    sourceIndex = indices[0];
+  } else {
+    const sourceExists = await esClient.indices.exists({ index: sourceIndex });
+
+    if (!sourceExists) {
+      return res.boom.badRequest(`Source index ${sourceIndex} does not exist.`);
+    }
+
+    if (indices.includes(sourceIndex) === false) {
+      return res.boom.badRequest(`Source index ${sourceIndex} is not aliased with alias ${aliasName}.`);
+    }
+  }
+
+  if (!destIndex) {
+    const date = new Date();
+    destIndex = `cumulus-${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
+  }
+
+  const destExists = await esClient.indices.exists({ index: destIndex });
+
+  if (destExists) {
+    return res.boom.badRequest(`Destination index ${destIndex} exists. Please specify an index name that does not exist.`);
+  }
+
+  // create destination index
+  await esClient.indices.create({
+    index: destIndex,
+    body: { mappings }
+  });
+
+  log.info(`Created destination index ${destIndex}.`);
+
+  // reindex
+  esClient.reindex({
+    body: {
+      source: { index: sourceIndex },
+      dest: { index: destIndex }
+    }
+  });
+
+  const message = `Reindexing to ${destIndex} from ${sourceIndex}. Check the reindex-status endpoint for status.`;
+
+  return res.status(200).send({ message });
+}
+
+async function reindexStatus(req, res) {
+  const esClient = await Search.es();
+
+  const reindexTaskStatus = await esClient.tasks.list({ actions: ['*reindex'] });
+
+  await esClient.indices.refresh();
+
+  const indexStatus = await esClient.indices.stats({
+    metric: 'docs'
+  });
+
+  const status = {
+    reindexStatus: reindexTaskStatus,
+    indexStatus
+  };
+
+  return res.send(status);
+}
+
+async function completeReindex(req, res) {
+  const deleteSource = req.body.deleteSource;
+  const aliasName = req.body.aliasName || defaultIndexAlias;
+  const sourceIndex = req.body.sourceIndex;
+  const destIndex = req.body.destIndex;
+
+  const esClient = await Search.es();
+
+  if (!sourceIndex || !destIndex) {
+    return res.boom.badRequest('Please explicity specify a source and destination index.');
+  }
+
+  if (sourceIndex === destIndex) {
+    return res.boom.badRequest('The source index cannot be the same as the destination index.');
+  }
+
+  const sourceExists = await esClient.indices.exists({ index: sourceIndex });
+
+  if (!sourceExists) {
+    return res.boom.badRequest(`Source index ${sourceIndex} does not exist.`);
+  }
+
+  const destExists = await esClient.indices.exists({ index: destIndex });
+
+  if (!destExists) {
+    return res.boom.badRequest(`Destination index ${destIndex} does not exist.`);
+  }
+
+  await esClient.indices.updateAliases({
+    body: {
+      actions: [
+        { remove: { index: sourceIndex, alias: aliasName } },
+        { add: { index: destIndex, alias: aliasName } }
+      ]
+    }
+  }).then(() => {
+    log.info(`Removed alias ${aliasName} from index ${sourceIndex} and added alias to ${destIndex}`);
+  }, (err) =>
```

Comment:
Is there any reason to prefer this error handling style over `.catch()` after .`then()`? I think we use `.catch()` throughout our code and it's a bit clearer when you're scanning code

Reply:
No reason, I had ported it over from some old code and didn't change it. It is updated now.